compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1177,8 +1187,7 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
                 record!(self.tables.codegen_fn_attrs[def_id] <- self.tcx.codegen_fn_attrs(def_id));
             }
             if should_encode_visibility(def_kind) {
-                let vis =
-                    self.tcx.local_visibility(local_id).map_id(|def_id| def_id.local_def_index);
+                let vis = self.tcx.visibility(def_id).map_id(|def_id| def_id.index);
```

Comment:
Why is this change needed?
If the visibility turns out non-local here it means a bug, it won't be encoded correctly.
`local_visibility` should always be preferred if it's possible.

Reply:
Because for newly synthesized associated types coming from RPITITs we would need to feed the visibility value using the query system. Otherwise we would execute local_visibility using the synthesized def_id and it will crash trying to access the non existing HIR.

Reply:
`TyCtxt::local_visibility` calls the `TyCtxt::visibility` query, though, and you're feeding that. If you revert this change, how does it ICE?

Reply:
Let's remove the change and we can introduce it later, some things are hard to justify now without any test.